### DISCUSSION TITLE
ui.switch not working to toggle the Dark Mode. Button does it.

### DISCUSSION THREAD

**Discussion by amitshipra** [zauberzeug/nicegui#3282] [Q&A] [answered]
### Question

```py
from nicegui import ui

dark = ui.dark_mode()
ui.label('Switch mode:')
light_mode = ui.switch("Dark Mode", value=False)
print(f'Light Mode: {light_mode.value}')
if light_mode.value:
    print('Enabling Light Mode')
    dark.disable
else:
    print('Enabling Dark Mode')
    dark.enable

ui.button('Dark', on_click=dark.enable)
ui.button('Light', on_click=dark.disable)

ui.run()
```

The dark.enable works when called form on_click but not otherwise. 

Novice question.

**Comment by python-and-novella** [comment]
Your code is confused. dark's `enable` and `disable` are function,should be called like `dark.enable()`.
You can bind the `value` of `dark_mode` with `switch`.
```python3
from nicegui import ui

dark = ui.dark_mode()
ui.label('Switch mode:')
light_mode = ui.switch("Dark Mode", value=False).bind_value(dark,'value')

ui.run()
```

**Comment by falkoschindler** [comment, marked as answer]
Hi @amitshipra,

As far as I can tell, there are three problems with your code:

1. As @python-and-fiction mentioned, you need to _call_ `dark.enable()` and `dark.disable()` to change anything.
2. Instead of running the if-statement only once after creating the UI elements, you should run it every time the switch is toggled. This can be done using a change event handler. (In [Streamlit](https://streamlit.io/) your approach would actually work because of their very unconventional way of interpreting Python code. In contrast, NiceGUI behaves like "normal" Python programs in this regard.)
3. Since the switch has the label "Dark Mode", I'd expect it to _enable_ dark mode when active. Therefore I'd swap the if and else blocks.

```py
def handle_switch():
    print(f'Light Mode: {switch.value}')
    if switch.value:
        print('Enabling Dark Mode')
        dark.enable()
    else:
        print('Enabling Light Mode')
        dark.disable()

dark = ui.dark_mode()
switch = ui.switch("Dark Mode", value=False, on_change=handle_switch)
```
